### PR TITLE
Start shaving off expression times

### PR DESCRIPTION
**Pull request by stephencelis**
Got a lot of things that exponentially affect compile time :(

### CODE REVIEW

File: Development.xcconfig [hunk 1]
```diff
@@ -0,0 +1 @@
+OTHER_SWIFT_FLAGS=$(inherited) -Xfrontend -warn-long-expression-type-checking=500
```
Comment (stephencelis) [inline Development.xcconfig]: Starting with 500...

File: Sources/PointFree/Account/Cancel.swift [hunk 2]
```diff
@@ -9,18 +9,25 @@ import Prelude
 import Styleguide
 @testable import Tuple
 
+func requireSubscriptionAndOwner<A>(
```
Comment (stephencelis) [inline Sources/PointFree/Account/Cancel.swift]: All of the middleware below took several seconds to compile before extracting these shared filters. Now it's under 500 ms.

File: Sources/PointFree/Admin/Index.swift [hunk 3]
```diff
@@ -62,18 +64,15 @@ private let newEpisodeEmailRowView = View<Episode> { ep in
     ])
 }
 
-let sendNewEpisodeEmailMiddleware =
-  filterMap(require1 >>> pure, or: loginAndRedirect)
-    <<< requireAdmin
-    <<< filterMap(fetchEpisode >>> pure, or: redirect(to: .admin(.newEpisodeEmail(.show))))
-    <| { conn in pure(conn.map(get2)) }
-    >-> sendNewEpisodeEmails
+let sendNewEpisodeEmailMiddleware:
```
Comment (stephencelis) [inline Sources/PointFree/Admin/Index.swift]: This was taking 20 seconds to compile for me :(

I cleaned up some of the filtering logic and extracted some other stuff and now it's under 500 ms.

File: Sources/PointFree/Emails/Mailgun.swift [hunk 4]
```diff
@@ -47,20 +47,17 @@ public struct SendEmailResponse: Decodable {
...
-    "subject": email.subject,
-    "text": email.text,
-    "html": email.html,
-    "tracking": email.tracking?.rawValue,
-    "tracking-clicks": email.trackingClicks?.rawValue,
-    "tracking-opens": email.trackingOpens?.rawValue
-    ]
-    |> compact
-
+  var params: [String: String] = [:]
```
Comment (stephencelis) [inline Sources/PointFree/Emails/Mailgun.swift]: A lil mutation never hurt anyone. 9 seconds to sub-500 ms.

File: Sources/PointFree/LaunchSignup/Views.swift [hunk 5]
```diff
@@ -153,18 +153,15 @@ let responsiveSuccessStyle =
     <> bottom(0)
 
 let responsiveStyle = queryOnly(screen, [minWidth(.px(800))]) {
-  ".hero, .signup, .success" % (
```
Comment (stephencelis) [inline Sources/PointFree/LaunchSignup/Views.swift]: Another 9ish second bit of code down to under 500 ms.

File: Sources/PointFree/Admin/Index.swift [hunk 6]
```diff
@@ -15,19 +15,22 @@ private let adminEmails = [
...
-  ) -> Middleware<StatusLineOpen, ResponseEnded, T2<Database.User, A>, Data> {
+  )
+  -> Middleware<StatusLineOpen, ResponseEnded, T2<Database.User?, A>, Data> {
 
-  return { conn in
-    conn
-      |> (adminEmails.contains(get1(conn.data).email.unwrap) ? middleware : redirect(to: .secretHome))
-  }
+    return filterMap(require1 >>> pure, or: loginAndRedirect)
+      <<< filter(get1 >>> isAdmin, or: redirect(to: .secretHome))
```
Comment (stephencelis) [inline Sources/PointFree/Admin/Index.swift]: A new `filter` :D

File: Sources/PointFree/TODO.swift [hunk 7]
```diff
@@ -125,6 +125,16 @@ public func filterMap<A, B>(
     }
 }
 
+public func filter<A>(
```
Comment (stephencelis) [inline Sources/PointFree/TODO.swift]: Was using this on another branch and wanted to use it to clean up above!

**Comment by stephencelis** [comment]
Build times have quite a range but this first commit shaves off about 2 min of build time!

image

image

File: Sources/PointFree/Footer.swift [hunk 8]
```diff
@@ -8,15 +8,23 @@ let footerView =
   curry(footer)([`class`([footerClass])]) >>> pure
     <¢> footerInfoColumnsView
 
+private func column(sizes: [Breakpoint: Int]) -> ([Node]) -> [Node] {
+  return gridColumn(sizes: sizes) >>> pure
```
Comment (stephencelis) [inline Sources/PointFree/Footer.swift]: Been finding that `pure` inference is having a tough time. Should talk about making them `static`s in Prelude.

File: Sources/PointFree/Account/Cancel.swift (hunk 2 above)
Comment (mbrandonw) [reply Sources/PointFree/Account/Cancel.swift]: dang. well at least its nice to see how easy it is to define new transformers out of existing!

File: Sources/PointFree/Admin/Index.swift (hunk 6 above)
Comment (mbrandonw) [reply Sources/PointFree/Admin/Index.swift]: nice! I like!

File: Sources/PointFree/Footer.swift (hunk 8 above)
Comment (mbrandonw) [reply Sources/PointFree/Footer.swift]: yeah makes sense to me. could start with code gen'ing them both. think we should figure out a code gen sitch for us anyway.

File: Sources/PointFree/LaunchSignup/Views.swift (hunk 5 above)
Comment (mbrandonw) [reply Sources/PointFree/LaunchSignup/Views.swift]: yeeeesh

File: Development.xcconfig [hunk 9]
```diff
@@ -0,0 +1 @@
+OTHER_SWIFT_FLAGS=$(inherited) -Xfrontend -warn-long-expression-type-checking=400
```
Comment (mbrandonw) [inline Development.xcconfig]: does `400` give us mostly no warnings?

Comment (stephencelis) [reply Development.xcconfig]: Still about 10 or so :(

Comment (mbrandonw) [reply Development.xcconfig]: Ah I was gonna say we should raise it till the warnings go away and then we can lower if we wanna do another pass later

Comment (stephencelis) [reply Development.xcconfig]: Ah, it's not too noisy at 400, though, and it only shows up for `make xcodeproj`. Maybe it'll motivate us to optimize a few along the way :)

**Pull request merged**